processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/UnivocityCsvParser.java

```diff
@@ -102,8 +102,8 @@ public void initialize() throws IOException {
     parserSettings.setMaxColumns(
         getMaxColumnsForParsing(csvParserVo.getNumberOfColumns(), csvParserVo.getMaxColumns()));
     parserSettings.setNullValue("");
-    parserSettings.setIgnoreLeadingWhitespaces(false);
-    parserSettings.setIgnoreTrailingWhitespaces(false);
+    parserSettings.setIgnoreLeadingWhitespaces(csvParserVo.getTrim());
```

Comment:
Also one more point it will be better to set this property in column level while creating the table itself as its column properties , this will avoid user to provide this option every time while data loading